### PR TITLE
[Bug 1635985] Import Docker Worker

### PR DESCRIPTION
**Pull request by imbstack**
~I'm a bit annoyed this is giving me `committed` credit for every rev so I'm going to go back and try to fix that.~

Otherwise next TODOs are around bringing over `.taskcluster.yml` tasks among other things I expect. Of particular interest to review is my manual lint fixes. (See notes below)

Suggestions of what to do next and ideas about what I've done wrong here would be much appreciated.

### CODE REVIEW

**Comment by walac** [comment]
I see a lot of replacements of `for (let x in` by `for (let x of`, but didn't see any change to the `for` body. Doesn't it change the semantics of the code?

**Comment by imbstack** [comment]
That change came from [guard-for-in](https://eslint.org/docs/rules/guard-for-in). I tried to be careful about making the switch. I think in each of these cases we are safe to make the change but good question, I will go through and try to make sure.

**Comment by imbstack** [comment]
Ok, the only step that needs to happen now (I think) is to update the image from `docker-worker/test/images/ci` to work in a a `version-1` tc-gh context. It assumes lots of env-vars and such currently.

Otherwise I think we should pretty much just land this as-is to unblock other work on integrating it better into the monorepo. Does that make sense to you, @walac what else would there be to do?

One note (to my future self mostly) is that this should not be landed yet until the commit `Temporarily add d-w tasks to PRs ` has been undone.

File: workers/docker-worker/src/lib/gc.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ GarbageCollector.prototype = {
     let containers = await this.docker.listContainers({all: true});
     let runningImages = containers.map((container) => { return container.Image.replace(/:latest$/, ''); });
 
-    for (let image in this.markedImages) {
+    for (let image of this.markedImages) {
       if (!exceedsThreshold && this.markedImages[image] > new Date()) {
```
Comment (walac) [inline workers/docker-worker/src/lib/gc.js:150]: Isn't `image` now the contant of the old `this.markedImages[images]`?

File: .taskcluster.yml [hunk 2]
```diff
@@ -413,6 +413,228 @@ tasks:
               - queue:scheduler-id:taskcluster-github
               - queue:create-task:highest:proj-taskcluster/gw-ci-*
               - queue:route:checks
+          dockerWorkerTasks: # These remain separate from workerTasks for now until we audit permissions (bug 1635985)
```
[the remark this answers is not in the document]
Comment (imbstack) [reply .taskcluster.yml:416]: I've disabled the tests for now and we'll enable them as a followup!

**Comment by imbstack** [comment]
Ok, I've gotten the tests to run in vagrant locally from this setup. It's a bit red but I think the failures are not from the lint fixes, etc.
I had a pass over and I think I fixed the for...of stuff but I still worry a bit so maybe I'll go back through again after we land it and check once I get tests more green.

**Comment by imbstack** [comment]
Here goes nothing!

**Pull request merged**